core/downloadfromdcthread.cpp

```diff
@@ -64,6 +65,27 @@ void DownloadThread::run()
 	dcs->set_device_name(m_data->devBluetoothName());
 }
 
+static const QVector<dc_family_t> ftdiFamilies = {
+	DC_FAMILY_SUUNTO_SOLUTION, DC_FAMILY_SUUNTO_EON, DC_FAMILY_SUUNTO_VYPER, DC_FAMILY_SUUNTO_VYPER2, DC_FAMILY_SUUNTO_D9,
+	DC_FAMILY_OCEANIC_VTPRO, DC_FAMILY_OCEANIC_VEO250, DC_FAMILY_OCEANIC_ATOM2,
+	DC_FAMILY_HW_OSTC, DC_FAMILY_HW_OSTC3, // but careful, the OSTC 4 is DC_FAMILY_HW_OSTC3 but has no serial connector
+	DC_FAMILY_CRESSI_EDY, DC_FAMILY_CRESSI_LEONARDO,
+	DC_FAMILY_DIVERITE_NITEKQ,
+	DC_FAMILY_DIVESYSTEM_IDIVE,
+	DC_FAMILY_TECDIVING_DIVECOMPUTEREU};
+
+bool guessIfFTDI(dc_descriptor_t *descriptor)
+{
+	if (!descriptor)
+		return false;
+	dc_family_t family = dc_descriptor_get_type(descriptor);
+	if (family == DC_FAMILY_HW_OSTC3 && same_string(dc_descriptor_get_product(descriptor), "OSTC 4"))
```

Comment:
There are other devices in the OSTC3 family which are bluetooth.

Maybee you should filter out those with DC_TRANSPORT_BLUETOOTH or DC_TRANSPORT_BLE?

Reply:
I guess the function name could change... this only gets called for descriptors that we know include serial transport...
But that reminds me, doesn't @jefdriesen consider BT/rfcomm a serial transport? So this may not be sufficient after all now that I think about it...

Reply:
Dive computers supporting rfcomm will indeed report both DC_TRANSPORT_BLUETOOTH and DC_TRANSPORT_SERIAL. The serial transport is there to indicate support for the serial port emulation mode. There are still many users of that feature, especially on Windows. In some cases it's even the only option. On Windows there exist several alternative bluetooth stacks (e.g. Widcomm and Toshiba) besides the Microsoft stack. But the Windows rfcomm api supports only the Microsoft stack. So if a user happens to have one of the alternative stacks installed, then the serial port emulation mode is the only one that will work.